src/test/java/hudson/plugins/git/GitSCMTest.java

```diff
@@ -143,6 +149,30 @@ private StandardCredentials getInvalidCredential() {
         return new UsernamePasswordCredentialsImpl(scope, id, "desc: " + id, username, password);
     }
 
+    @Issue("JENKINS-56404")
+    @Test
+    public void testRedundantFetch() throws Exception {
```

Comment:
While the first time checkout would work, could you write a case which tests whether `retrieveChanges` is able to fetch an update?

Reply:
I will try to do it after the changes made.